jade_client/src/test/scala/org/broadinstitute/monster/ingester/jade/JadeApiClientSpec.scala

```diff
@@ -0,0 +1,238 @@
+package org.broadinstitute.monster.ingester.jade
+
+import java.time.OffsetDateTime
+import java.time.format.DateTimeFormatter
+
+import cats.effect.{IO, Resource}
+import fs2.Stream
+import io.circe.Json
+import org.broadinstitute.monster.ingester.jade.JadeApiClient.{
+  JadeApiStatusEndpoint,
+  JadeApiJobStatusEndpoint,
+  JadeApiIngestEndpoint
+}
+import org.broadinstitute.monster.ingester.jade.models.ApiError.JadeError
+import org.broadinstitute.monster.ingester.jade.models.JobStatus.Running
+import org.broadinstitute.monster.ingester.jade.models.{
+  ApiErrorBody,
+  IngestRequest,
+  JadeStatus,
+  JobInfo
+}
+import org.http4s._
+import org.http4s.circe.CirceEntityDecoder._
+import org.scalatest.{FlatSpec, Matchers}
+
+class JadeApiClientSpec extends FlatSpec with Matchers {
+  // So we likely want a simple mock of the Jade API or something like that. Gonna try to piggyback off the GCS stuff
+
+  private val datasetId = "datasetid"
+  private val jobId = "jobid"
+  private val datasetData = IngestRequest("path", "table")
+
+  private def buildApi(run: Request[IO] => Resource[IO, Response[IO]]): JadeApiClient =
+    new JadeApiClient(run)
+
+  behavior of "JadeApiClient"
+
+  // ingest
+  it should "return a properly formatted response of JobInfo when an ingest request is successfully requested" in {
+    val thetime = OffsetDateTime.now()
+    val thetimestring = thetime.format(DateTimeFormatter.ISO_OFFSET_DATE_TIME)
+    val api = buildApi { req =>
+      req.method shouldBe Method.POST
+
+      req.uri shouldBe JadeApiIngestEndpoint / datasetId / "ingest"
+
+      val expectedBody = Json.obj(
+        "path" -> Json.fromString("path"),
+        "table" -> Json.fromString("table"),
+        "format" -> Json.fromString("json"),
+        "max_bad_records" -> Json.fromInt(0)
+      )
+
+      val actualBody = req.as[Json].unsafeRunSync()
```

Comment:
Instead of `unsafeRunSync()`, since we're returning a `Resource` anyway:
```scala
Resource.liftF(req.as[Json]).map { actualBody =>
  actualBody shouldBe expectedBody
  Response[IO](...)
}
```